### ISSUE TITLE
Units fail on Titan

### ISSUE THREAD

**Issue by jrossyra** [radical-cybertools/radical.pilot#1511]
@itomaldonado On Titan I have the pilot submit a job successfully, but when the compute units come up they fail with an empty STDOUT, so it looks to me like the error happens from the `unit.X.sh` script. I am not absolutely certain of where the error comes from, however it seems like line 24 of this unit script is the culprit. 

I can upload more, here is one of the units (I checked that they are all the same), and all the files in the pilot directory.

[pilot.0000.tar.gz](https://github.com/radical-cybertools/radical.pilot/files/1532016/pilot.0000.tar.gz)
[unit.000001.tar.gz](https://github.com/radical-cybertools/radical.pilot/files/1532012/unit.000001.tar.gz)

**Comment by vivek-bala** [comment]
I don't think that is the cause of the failure. Touch without arguments gives you an error/warning but its not critical. I see the following line:

```
/sw/xk6/xalt/0.7.5/bin/aprun -L 2348 -n 1 /lustre/atlas/proj-shared/bip149/jrossyra/admdrpenv/bin/activate
```

There is no executable mentioned, just the arguments. Right?

**Comment by jrossyra** [comment]
I had a previous error just before this one came up, where my virtualenv activate script was not executable. I changed the permissions to make that go away, and left with the impression that this was the executable for aprun.

**Comment by vivek-bala** [comment]
I think the executable should be 'source' and arguments should be the link to your virtualenv. But more than that, 'source <VE>' is probably something to be put in the pre_exec and the executable+arguments should be something compute intensive. What do you think?

**Comment by andre-merzky** [comment]
I think there are two errors here: first, RP screws up by not setting `RP_PROF`, which leads to the `touch` message - I'll fix that.  But second I agree that the aprun command line look incomplete, and that the VE activation should be moved to the pre-exec part of the CU.

**Comment by andre-merzky** [comment]
The `touch` problem is fixed in branch `fix/issue_1511`, and will go into release `0.48` (planned for later this year).
